lib/carrierwave/downloader/remote_file.rb

```diff
@@ -18,6 +18,8 @@ def initialize(file)
           @headers = file.meta
           @uri = file.base_uri
         end
+      rescue StandardError => e
+        raise CarrierWave::DownloadError, "could not download file: #{e.message}"
```

Comment:
Why is this change?

Reply:
There is another bug in ssrf_filter 1.1 that is not fixed by your PR, which sometimes causes it to fail to retrieve the body (or maybe when it failed to retrieve the body), without letting the Net::HTTPResponse `reading_body` method call `self.body` in its ensure clause, and *not raising an exception*. Then, when `file.body` is called above outside of the `reading_body` method we get that `stream_check` error people were mentioning in the issue.

I don't know what causes this, but I definitely have seen Rollbar reports of this happening on our servers after applying the patch for the `request_proc` problem (I tried both your patch and mine, separately), and it still happened, and I don't know why. Only downgrading ssrf_filter made that problem go away.

Regardless of whether you PR gets merged, this change is definitely needed. Given the tools used, we can't assume that all the exceptions will be raised before you get to this file.

For one thing, if the response of the download was 204, then `file.body` will be nil, and `StringIO.new(file.body)` will raise a TypeError. At the very least, if you don't want to make a distinction between no body and and an empty body, that should be changed to `StringIO.new(file.body.to_s)` so we don't raise an exception on a condition that definitely can happen normally.

If I get the time, I'll try to track down the bug in ssrf_filter that causes these `stream_check` errors in the first place. I already made a PR to net-http to fix the NoMethodError on nil.

Reply:
Please do not make a workaround. If you think that we need to deal with the situation of '204 No Content', please do so in a separate PR.
Raising `CarrierWave::DownloadError` results in a validation error, but users can't do anything on getting errors like https://github.com/arkadiyt/ssrf_filter/issues/58.

Reply:
Removed the rescue code, per request. Made #2633 for the no content responses.